nix/nixos/cardano-node-service.nix

```diff
@@ -39,37 +49,43 @@ let
           "--shelley-operational-certificate ${cfg.operationalCertificate}"}"
       ];
     };
+    instanceDbPath = "${cfg.databasePath}${optionalString (i > 0) "-${toString i}"}";
     exec = "cardano-node run";
         cmd = builtins.filter (x: x != "") [
           "${cfg.package}/bin/${exec}"
           "--config ${realNodeConfigFile}"
-          "--database-path ${cfg.databasePath}"
-          "--topology ${cfg.topology}"
+          "--database-path ${instanceDbPath}"
+          "--topology ${topology}"
+        ] ++ (lib.optionals (!cfg.systemdSocketActivation) [
           "--host-addr ${cfg.hostAddr}"
           "--port ${toString cfg.port}"
-        ] ++ lib.optional (!cfg.systemdSocketActivation) "--socket-path ${cfg.socketPath}"
+          "--socket-path ${cfg.socketPath}"
+        ])
           ++ consensusParams.${cfg.nodeConfig.Protocol} ++ cfg.extraArgs ++ cfg.rtsArgs;
     in ''
         choice() { i=$1; shift; eval "echo \''${$((i + 1))}"; }
         echo "Starting ${exec}: ${concatStringsSep "\"\n   echo \"" cmd}"
         echo "..or, once again, in a single line:"
         echo "${toString cmd}"
         ${lib.optionalString (cfg.environment == "selfnode") ''
-          echo "Wiping all data in ${cfg.stateDir}"
-          rm -rf ${cfg.stateDir}/*
+          echo "Wiping all data in ${instanceDbPath}"
+          rm -rf ${instanceDbPath}/*
           GENESIS_FILE=$(${pkgs.jq}/bin/jq -r .GenesisFile < ${cfg.nodeConfigFile})
           START_TIME=$(date +%s --date="30 seconds")
           ${pkgs.jq}/bin/jq -r --arg startTime "''${START_TIME}" '. + {startTime: $startTime|tonumber}' < $GENESIS_FILE > ${cfg.stateDir}/genesis.json
```

Comment:
${cfg.stateDir} should also be ${instanceDbPath} in this line?

Reply:
actually `selfnode` env is not supported with multi-instance for now so I'm going to bother doing adaptation for it (reverting above change).